maven/rules.bzl

```
@@ -315,6 +316,10 @@ def _deploy_maven_impl(ctx):
     src_jar_link = "lib.srcjar"
     pom_xml_link = ctx.attr.target[MavenDeploymentInfo].pom.basename
 
+    release_repo = ctx.attr.release
+    if _DEPLOY_MAVEN_RELEASE_REPO_KEY in ctx.var:
+        release_repo = ctx.var[_DEPLOY_MAVEN_RELEASE_REPO_KEY]
```

I agree with the approach of using `--define`! This is aligned with how we override the `version` being deployed:
```sh
bazel run --define version=$(cat VERSION) //:deploy-maven -- release
```
A couple of points:
1. Since we're allowing users to override the release repo, we should also allow them to override the snapshot repo.
2. As `version` is all lowercase, so should be `deploy_maven_release_repo`; and for terseness, I'd call it just `release_repo`. (As this is a `run` target, the key need not be globally unique.)

With all that in mind, and simplifying the Starlark a little:
```suggestion
    snapshot_repo = ctx.var.get("snapshot_repo", default=ctx.attr.snapshot)
    release_repo = ctx.var.get("release_repo", default=ctx.attr.release) 
```
(requires additional update to L330: `"{snapshot}": snapshot_repo,`)